tests/Kernel/Paragraphs/MediaParagraphsTest.php

```diff
@@ -737,4 +765,92 @@ public function testBanner(): void {
     );
   }
 
+  /**
+   * Test contact paragraph rendering.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function testContact(): void {
+    // Create a media entity for contact.
+    $file = file_save_data(file_get_contents(drupal_get_path('theme', 'oe_theme') . '/tests/fixtures/example_1.jpeg'), 'public://example_1.jpeg');
+    $file->setPermanent();
+    $file->save();
+    $media = $this->container->get('entity_type.manager')->getStorage('media')->create([
+      'bundle' => 'image',
+      'name' => "Test image",
+      'oe_media_image' => [
+        'target_id' => (int) $file->id(),
+        'alt' => "Alternative text",
+      ],
+      'uid' => 0,
+      'status' => 1,
+    ]);
+    $media->save();
+    // Create general contact.
+    $general_contact = Contact::create([
+      'bundle' => 'oe_general',
+      'name' => 'General contact',
+      'oe_address' => [
+        'country_code' => 'BE',
+        'locality' => 'Brussels',
+        'address_line1' => 'Address of General contact',
+        'postal_code' => '1001',
+      ],
+      'oe_body' => 'General contact body text',
+      'oe_email' => 'general@example.com',
+      'oe_organisation' => 'General contact Organisation',
+      'oe_social_media' => [
+        [
+          'uri' => 'http://www.example.com/facebook',
+          'title' => 'Facebook',
+          'link_type' => 'facebook',
+        ],
+      ],
+      'oe_image' => [
+        [
+          'target_id' => (int) $media->id(),
+          'caption' => "General contact media caption",
+        ],
+      ],
+      'status' => CorporateEntityInterface::PUBLISHED,
+    ]);
+    $general_contact->save();
+    // Create press contact.
+    $press_contact = Contact::create([
+      'bundle' => 'oe_press',
+      'name' => 'Press contact',
+      'oe_body' => 'Press contact body text',
+      'status' => CorporateEntityInterface::PUBLISHED,
+    ]);
+    $press_contact->save();
+    // Create contact paragraph.
+    $paragraph = Paragraph::create([
+      'type' => 'oe_contact',
+      'field_oe_contacts' => [$general_contact, $press_contact],
+    ]);
+    $paragraph->save();
+    $html = $this->renderParagraph($paragraph);
+    $crawler = new Crawler($html);
+
+    // Assert paragraph heading markup is not rendered if no title is set.
+    $this->assertCount(0, $crawler->filter('h2.ecl-u-type-heading-2'));
```

Comment:
Isnt there something we can borrow from ContactRenderTest to reuse some of the assertions? This seems quite duplicate since the full view mode is tested elsewhere

Reply:
ContactRenderTest tests in detail each part of the contact being rendered in full view mode and would help only for asserting the featured media. In the paragraph test we only test that contact is being rendered using the full view mode.